iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -665,7 +665,7 @@ private async Task SubscribeCloudToDeviceMessagesAsync()
         {
             if (TryStateTransition(TransportState.Open, TransportState.Subscribing))
             {
-                await _channel.WriteAsync(new SubscribePacket(0, new SubscriptionRequest(_deviceboundMessageFilter, QualityOfService.AtMostOnce))).ConfigureAwait(true);
+                await _channel.WriteAsync(new SubscribePacket(0, new SubscriptionRequest(_deviceboundMessageFilter, QualityOfService.AtLeastOnce))).ConfigureAwait(true);
 
```

Comment:
What would be the impact on the QOS of messages to SDK users?

Reply:
We were initially subscribing with QoS of 0, so all C2D messages were sent from service with QoS of 0 -> at most once -> fire-and-forget. We now subscribe with QoS of at-least-once, so this will make c2d message delivery more reliable over Mqtt.